### PR TITLE
Refactoring

### PR DESCRIPTION
**Pull request by cpetschnig**
This is supposed to change no functionality at all. Basically, some methods are extracted into a new class.

This refactoring is needed for the v3 extension, that changes the signing algorithm.

### CODE REVIEW

File: spec/sageone_api_signer_spec.rb [hunk 1]
```diff
@@ -53,70 +58,38 @@
...
-    end
-
-    describe 'with a specific port' do
-      before { subject.url = 'https://api.sageone.com:123/accounts/v1/contacts?config_setting=foo' }
-      it { expect(subject.base_url).to eql 'https://api.sageone.com:123/accounts/v1/contacts' }
+  describe '#request_method' do
+    it 'BUG nil the second time we call it!!!' do
+      subject.request_method = 'get'
+      expect(subject.request_method).to eql 'GET'
+      expect(subject.request_method).to eql 'GET'
```
Comment (timlapluie) [inline spec/sageone_api_signer_spec.rb:73]: This line is duplicated.

[EDIT]
Just see my comment as a true statement with no need of changes. :wink:

Comment (cpetschnig) [inline spec/sageone_api_signer_spec.rb:73]: I also deleted it twice, so it sums up to zero ;)

Comment (timlapluie) [inline spec/sageone_api_signer_spec.rb:73]: :+1:

**Comment by LeFnord** [comment]
:+1:

**Comment by timlapluie** [comment]
:+1:

**Pull request merged**